src/indexable.cr

```diff
@@ -23,34 +23,32 @@ module Indexable(T)
   # This method should only be directly invoked if you are absolutely
   # sure the index is in bounds, to avoid a bounds check for a small boost
   # of performance.
-  abstract def unsafe_at(index : Int)
+  abstract def unsafe_fetch(index : Int)
 
   # Returns the element at the given *index*, if in bounds,
-  # otherwise executes the given block and returns its value.
+  # otherwise executes the given block with the index and returns its value.
   #
   # ```
-  # a = [:foo, :bar]
-  # a.at(0) { :baz } # => :foo
-  # a.at(2) { :baz } # => :baz
+  # a = [1, 2]
+  # a.fetch(1) { |index| index * 3 } # => 6
+  # a.fetch(3) { |index| index * 3 } # => 9
```

Comment:
This new example code is IMHO less clear since it's harder to recognise at the first sight what's key and what's value, previous one was more obvious in that.

Reply:
And it's incorrect. `a.fetch(1) { |index| index * 3 }` would return the element at index 1 which is 2.

Reply:
> This new example code is IMHO less clear since it's harder to recognise at the first sight what's key and what's value, previous one was more obvious in that (ditto below).

OK, I agree, changed to multiplies of ten.

> And it's incorrect. a.fetch(1) { |index| index * 3 } would return the element at index 1 which is 2.

Yes, thank you.

Reply:
> OK, I agree, changed to multiplies of ten.

You missed the point, `Symbol` values were way more clear.